server/public/model/client4.go

```diff
@@ -3032,6 +3036,26 @@ func (c *Client4) CreateChannel(ctx context.Context, channel *Channel) (*Channel
 	return ch, BuildResponse(r), nil
 }
 
+func (c *Client4) GetOrCreateDirectChannel( ctx context.Context, otherUserID string) (*Channel, *Response, error){
```

Comment:
This is how I defined the GetOrCreateDirectChannel function in Client4 but when I build mmctl and execute the command

`mmtcl post create dm "Hey there!"`

I get this error on the image below.

![Screenshot from 2024-09-16 18-20-22](https://github.com/user-attachments/assets/75a8ecae-c6fc-4589-8a7f-4074c5db699d)


Error indicates that there is no such url as `/api/v4/channels/direct` meanwhile it is the URL for hitting the create direct message endpoint.

Is there anything I am not doing correctly or something I need to add?

Reply:
Can you use the existing `CreateDirectChannel` method? Or does that one return an error if the direct channel already exists?

Reply:
> Can you use the existing `CreateDirectChannel` method? Or does that one return an error if the direct channel already exists?

I checked but the Client interface in `mmctl/client/client.go` does not have `CreateDirectChannel` method that is why I tried to add `GetOrCreateChannel` which should of course do what `CreateDirectChannel` method would do if it existed on the `Client` interface

Reply:
You can add the `CreateDirectChannel` method to `mmctl/client/client.go`, but you don't have to implement it. The implementation already exists in `public/model/client4.go`.

Reply:
Okay, I will verify sir

Reply:
Hello @hanzei  I have added `CreateDirectChannel` to mmctl client as you requested  but I still get the same error.
These a logs information related to the command execution. Maybe it can be of help :point_down: 
![Screenshot from 2024-09-17 23-13-45](https://github.com/user-attachments/assets/3b7b2776-0440-472a-89bd-2d3696418f7c)

Reply:
Good day @hanzei can you have a look at the issue with the DoAPIPost method please?
Even `mmctl post create` command is failing to create a post in a channel. With thesame error message as above that an endpoint with url `/api/v4/posts` does not exist

Reply:
Please excuse the delay. I think I found the issue. Please take a look at https://github.com/mattermost/mattermost/pull/26602/files#r1777055106

Reply:
Okay let me look at it

Reply:
So do I need to add the query parameter `set_online=false`?

Reply:
The problem is that you are passing an empty string to `CreateDirectChannel`.